### DISCUSSION TITLE
vipsthumbnail speed of different versions / Win

### DISCUSSION THREAD

**Discussion by flpxra** [libvips/libvips#3006] [Q&A] [answered]
I wonder about libvips speed over many versions. 
I am using the Windows version (`vips-dev-w64-web`) and the command:
`vipsthumbnail.exe IMG_0128.jpg --size "x800" -o out.jpg `

Probably my tests are not 100% reliable but I tested it on various files, repeating it many times etc. The results always show that version 8.10.5 is the best.

What is the difference, and what are the chances that the newer versions will achieve at least the same as version 8.10.5?

Results:
```
8.13.0 - 	0.108 s
8.12.2 - 	0.107 s
8.11.3 - 	0.103 s
8.10.6 - 	0.105 s
8.10.5b - 	0.098 s
8.10.5 - 	0.096 s
8.10.2 - 	0.107 s
8.9.2 - 	0.147 s
8.9.0 - 	0.145 s
8.8.4 - 	0.139 s
8.8.1 - 	0.139 s
8.7.3 - 	0.153 s
8.7.2 - 	0.153 s
```

**Comment by jcupitt** [comment, marked as answer]
Hi @flexphperia,

It's a combination of factors. Over time, we've improved performance (increases speed), improved quality (often decreases speed), and improved our build process (can go either way). And of course the projects we depend on, like libjpeg, have evolved as well.

We're looking at adopting Highway, which ought to give a speed boost, but it's not ready yet.
